interfaces/builtin/utils.go

```diff
@@ -110,7 +111,20 @@ func aareExclusivePatterns(orig string) []string {
 // dirs.SnapDesktopFilesDir, but explicitly denies access to all other snaps'
 // desktop files since xdg libraries may try to read all the desktop files
 // in the dir, causing excessive noise. (LP: #1868051)
-func getDesktopFileRules(snapInstanceName string) []string {
+func getDesktopFileRules(snapInstanceName string, spec *apparmor.Specification) []string {
+	// desktop-launch allows to read all .desktop files; but "deny" rules overrule any "allow"
+	// rule, so we must not add these rules if this snap uses the desktop-launch interface.
+	// Also, for security reasons, all these rules are removed if the desktop-launch interface
+	// is listed, thus only if it is really connected will the snap have any kind of access to
+	// these folders/files.
+	if spec != nil {
+		for _, plug := range spec.SnapAppSet().Info().Plugs {
+			if plug.Interface == "desktop-launch" {
```

Comment:
I'll prepare that test.

About the future, what do you think about this approach? Let's say that I implement a method like this in the `spec` class:

    func (*spec Specification) AddSnippetWithPriority(snippet string, ID string, priority int) {
        ...
    }

that allows to add a snippet but specifying a priority. That way, both interfaces would add their snippet using the same ID, but desktop-launch would set a bigger priority than desktop-legacy. That function would check if there is already a snippet with the same ID, and in that case, if the priority is lower than the new one, in which case that snippet will be replaced.

I see two ways of keeping track of the snippets:

* the "quick and ugly one": add two comments with a specific format to specify the beginning and the ending of a snippet with priority, and containing the ID and the current priority. This would allow to replace the text in the current snippet.

* the "elegant but not-so-quick one": add a second dictionary that contains the snippets with priority, and modify all the methods that read the whole snippets to, instead, call a private method that gets the "normal" snippets and the prioritized ones and mix them in a single snippet.

What do you think?

Reply:
Added the extra test.

Reply:
thanks, for the test.  Yes, what you outline there is probably what we want to do up to some naming discussions, but definitely using the some variant of the "elegant but not-so-quick one" approach, that's why I'm ok landing this first

Reply:
Perfect. Then I'll prepare a little braindump document, to begin to discuss that. Thanks!